Workflow params (#5929)
nextflow-io/nextflow: docs/workflow.md

}
```

(workflow-params-def)=

## Parameters

> There are two ways to declare parameters in a Nextflow script: the `params` block and *legacy* parameters.

Reviewer: ```suggestion
Parameters in a Nextflow script can be declared in the `params` block and as *legacy* parameters.
```
Please check meaning is retained.

### Params block

:::{versionadded} 25.04.0
:::

:::{note}
This feature requires the `nextflow.preview.params` feature flag to be enabled.
:::

A script can declare parameters using the `params` block:

```nextflow
params {
    /**

>      * Path to input data.

Reviewer 2: what determines this is a path?

     */
    input

    /**
     * Whether to save intermediate files.
     */
    save_intermeds = false
}
```

Parameters can be used in the entry workflow:

```nextflow
workflow {
    if( params.input )
        analyze(params.input, params.save_intermeds)
    else
        analyze(fake_input(), params.save_intermeds)
}
```

:::{note}

> Named workflows should not use params. They should receive their inputs explicitly through the `take:` section.

Reviewer 2: Not sure about this. I think it should be maintained the use of parameters both in workflows and processes as should as they are immutable

Author: To be clear, parameters can still be used everywhere. This is just a recommendation since params cannot be validated outside the entry workflow. I can document this more explicitly if you want

Author: I have reframed this note as a best practice recommendation, since it is still allowed
